### ISSUE TITLE
Quote Item Prices are NULL in cart related events.

### ISSUE THREAD

**Issue by jakwinkler** [magento/magento2#18685]
### Preconditions (*)

1. Magento 2.1.15 EE

### Steps to reproduce (*)

1. Create a module to observer following events:

- `checkout_cart_product_add_after`
- `checkout_cart_product_update_after`

2. Try to get values of quote item price in observers:

```
    /**
     * @param \Magento\Framework\Event\Observer $observer
     */
    public function execute(\Magento\Framework\Event\Observer $observer)
    {
        /** @var \Magento\Quote\Model\Quote\Item $item */
        $item = $observer->getEvent()->getData('quote_item');
        $item->load($item->getId());
// part of my debug session
        var_dump(get_class($item));    // wil return correct class
        var_dump($item->getCustomPrice());  // will return NULL
        var_dump($item->getBasePrice());    // will return NULL
        die();
    }
```

### Expected result (*)

1. Quote Item Prices are loaded from database (they are set in `quote_item` table)
2. Be able to control product price on update action

### Actual result (*)

1. Quote Item prices are NULL
2. Can't control quote item price (setCustomPrice) becasue I don't know previous price (also customized one)

**Comment by magento-engcom-team** [comment]
Hi @qsolutions-pl. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@qsolutions-pl do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-95885`, `MAGETWO-95886` were created

**Comment by Rus0** [comment]
#MM18MX

**Comment by magento-engcom-team** [comment]
@Rus0 thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @ashutoshwebkul. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by ghost** [comment]
@Rus0 @magento-engcom-team this issue was already verified by me and it is also fixed in the these PR's:
https://github.com/magento/magento2/pull/18806
https://github.com/magento/magento2/pull/18808

**Comment by VladimirZaets** [comment]
Hi @qsolutions-pl. Thank you for your report. 
The issue has been fixed in magento/magento2#18808 by @ashutoshwebkul in 2.2-develop branch
Related commit(s):
 - [09586e4bbf2b95e0e7acd33c058f7daf2c798c63](https://github.com/magento/magento2/commit/09586e4bbf2b95e0e7acd33c058f7daf2c798c63)
 - [1638891a11aeab7673d6e0280f43ff864486406a](https://github.com/magento/magento2/commit/1638891a11aeab7673d6e0280f43ff864486406a)
 - [175ebc0bbc8cf2e4a5e4fabc59626cdfaf7aa519](https://github.com/magento/magento2/commit/175ebc0bbc8cf2e4a5e4fabc59626cdfaf7aa519)
 - [59bd87487dfc5f1ea69247b8b3ab7faee4ca486c](https://github.com/magento/magento2/commit/59bd87487dfc5f1ea69247b8b3ab7faee4ca486c)

The fix will be available with the upcoming 2.2.8 release.

**Comment by magento-engcom-team** [comment]
Hi @qsolutions-pl. Thank you for your report. 
The issue has been fixed in magento/magento2#23839 by @edenduong in 2.3-develop branch
Related commit(s):
 - [e77fc028bc2f22b8dfccd1961491b0c08318249a](https://github.com/magento/magento2/commit/e77fc028bc2f22b8dfccd1961491b0c08318249a)
 - [997a050518a374589f919073b6539b83ecaf3c02](https://github.com/magento/magento2/commit/997a050518a374589f919073b6539b83ecaf3c02)
 - [3c282f90d64fde86bf44c4b4b7e70548e493336a](https://github.com/magento/magento2/commit/3c282f90d64fde86bf44c4b4b7e70548e493336a)
 - [e14d5cce259e60d10817e03558c211755059e50b](https://github.com/magento/magento2/commit/e14d5cce259e60d10817e03558c211755059e50b)
 - [8b49ff98ca6203d8a38f8bb4054620c20dab686d](https://github.com/magento/magento2/commit/8b49ff98ca6203d8a38f8bb4054620c20dab686d)
 - [da18ecd570a8855bd2daa23ed6d6264c7fc603a5](https://github.com/magento/magento2/commit/da18ecd570a8855bd2daa23ed6d6264c7fc603a5)
 - [23bb9b5f44394e57ecbb3d603fd93e2533e98803](https://github.com/magento/magento2/commit/23bb9b5f44394e57ecbb3d603fd93e2533e98803)
 - [2b518365bbbb4d30d745b6a46536c3cad4ff7d50](https://github.com/magento/magento2/commit/2b518365bbbb4d30d745b6a46536c3cad4ff7d50)

The fix will be available with the upcoming 2.3.4 release.
